### PR TITLE
feat: Upgrade storybook to v7-latest for Webpack5

### PR DESCRIPTION
**Pull request by pasanchMSFT**
### Summary of changes 🔍 
Upgrade Storybook to V7-latest to use Webpack5. Webpack5 is recommended due to a high-risk security exploit detected for webpack-dev-middleware in <v5: https://github.com/advisories/GHSA-wr3j-pwj9-hqq6. 

Other changes include:
- ESLint rules for stories (most Story modifications come from this, https://github.com/storybookjs/eslint-plugin-storybook/blob/main/docs/rules/context-in-play-function.md)
- Removing parse-link, since browser polyfills are not supported in Webpack5. Alternative proposed here is a fork of that repo that is webpack5 compatible: https://www.npmjs.com/package/@web3-storage/parse-link-header.
- Included custom indexer for local stories: https://storybook.js.org/docs/api/main-config-indexers

### Testing 🧪
- Tested stories locally, commented out some that were having issues in Chromatic. Will investigate further later.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,177 +1,181 @@
...
-        "react-dom": ">=16.8.0"
-    },
-    "overrides": {
-        "minimist@1.2.5": "1.2.6"
-    },
-    "author": "",
-    "license": "ISC"
-}
\ No newline at end of file
+  "name": "@microsoft/iot-cardboard-js",
```
Comment (msnyder-msft) [inline package.json:2]: can we revert the formatting change so that we can see the actual diff?

**Review by msnyder-msft — COMMENTED** [review]
You've got a few other files that are flagged, is it related to your change?
image

**Review by msnyder-msft — APPROVED** [review]

**Review by rapotl — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (pasanchMSFT) [reply package.json]: Done

**Comment by pasanchMSFT** [comment]
> You've got a few other files that are flagged, is it related to your change? image

These seem to be lint suggestions from GH, unrelated to my changes.

**Comment by pasanchMSFT** [comment]
Both ValueRange stories and Ontology model selection stories work. Ontology authoring integration Story is currently broken because of an infinite loop, pending investigation.
RangeStoryWorks

OATStoryWorks

**Review by msnyder-msft — APPROVED** [review]

**Review by rapotl — APPROVED** [review]

**Pull request merged**